tests/api/core/strapi/api/sort/sort.test.api.ts

```diff
@@ -0,0 +1,403 @@
+        .filter((tag) => tag.name.endsWith('D') || tag.name.endsWith('A'))
+        .map((cat) => cat.id),
+    },
+    {
+      name: 'Category B',
+      locale: 'en',
+      tags: fixtures.tag.filter((tag) => tag.name.endsWith('C')).map((cat) => cat.id),
+    },
+    {
+      name: 'Category D',
+      locale: 'en',
+      tags: fixtures.tag.filter((tag) => tag.name.endsWith('A')).map((cat) => cat.id),
+    },
+  ],
+  tag: [{ name: 'Tag A' }, { name: 'Tag B' }, { name: 'Tag C' }, { name: 'Tag D' }],
+  articles(fixtures) {
+    return [
+      {
+        title: 'Article A',
+        locale: 'en',
+        categories: fixtures.category.filter((cat) => cat.name.endsWith('B')).map((cat) => cat.id),
+      },
+      {
+        title: 'Article C',
+        locale: 'en',
+        categories: fixtures.category.filter((cat) => cat.name.endsWith('D')).map((cat) => cat.id),
+      },
+      {
+        title: 'Article D',
+        locale: 'en',
+        categories: fixtures.category
+          .filter((cat) => cat.name.endsWith('A') || cat.name.endsWith('D'))
+          .map((cat) => cat.id),
+      },
+      {
+        title: 'Article B',
+        locale: 'en',
+        categories: fixtures.category
+          .filter((cat) => cat.name.endsWith('C') || cat.name.endsWith('D'))
+          .map((cat) => cat.id),
+      },
+    ];
+  },
+};
+
+const BASIC_ARTICLE_FIELDS = {
+  documentId: expect.any(String),
+  locale: 'en',
+  publishedAt: expect.anything(),
+  updatedAt: expect.anything(),
+  createdAt: expect.anything(),
+};
+
+/**
+ * Article(A) -> Categories(B)
+ * Article(B) -> Categories(C, D)
+ * Article(C) -> Categories(D)
+ * Article(D) -> Categories(A, D)
+ *
+ * Category(A) -> Tags(B)
+ * Category(B) -> Tags(C)
+ * Category(C) -> Tags(A, D)
+ * Category(D) -> Tags(A)
+ *
+ * 1 - Article(A) -> Categories(B) -> Tags(C)
+ * 4 - Article(B) -> Categories(C, D) -> Tags(A, D)
+ * 2 - Article(C) -> Categories(D) -> Tags(A)
+ * 3 - Article(D) -> Categories(A, D) -> Tags(A, B)
+ */
+describe('Sort', () => {
+  beforeAll(async () => {
+    await builder
+      .addContentTypes(Object.values(schemas.contentTypes))
+      .addFixtures(schemas.contentTypes.tag.singularName, fixtures.tag)
+      .addFixtures(schemas.contentTypes.category.singularName, fixtures.category)
+      .addFixtures(schemas.contentTypes.article.singularName, fixtures.articles)
+      .build();
+
+    strapi = await createStrapiInstance();
+    rq = createContentAPIRequest({ strapi });
+    data = await builder.sanitizedFixtures(strapi);
+  });
+
+  afterAll(async () => {
+    await strapi.destroy();
+    await builder.cleanup();
+  });
+
+  test('Regular sort', async () => {
+    const res = await rq.get(`/${schemas.contentTypes.article.pluralName}`, {
+      qs: { sort: 'title' },
+    });
+
+    expect(res.status).toBe(200);
+    expect(res.body.data.length).toBe(4);
+
+    expect(res.body.data).toMatchObject([
+      { id: 1, title: 'Article A', ...BASIC_ARTICLE_FIELDS },
```

Comment:
to avoid unpredictability you can remove the hardcoded ids from the matchers

Reply:
Good catch